sdk-base/src/main/java/com/mapbox/maps/extension/observable/ObservableExtension.kt

```diff
@@ -301,124 +299,333 @@ fun ObservableInterface.unsubscribeSourceRemoved(observer: Observer) =
-  val json = data.toJson()
-  return gson.fromJson(json, SourceAddedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return SourceAddedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for source removed event.
  * @return a parsed SourceRemovedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getSourceRemovedEventData(): SourceRemovedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, SourceRemovedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return SourceRemovedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for render frame started event.
  * @return a parsed RenderFrameStartedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getRenderFrameStartedEventData(): RenderFrameStartedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, RenderFrameStartedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return RenderFrameStartedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END)
+  )
 }
 
 /**
  * Get the parsed event data for render frame finished event.
  * @return a parsed RenderFrameFinishedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getRenderFrameFinishedEventData(): RenderFrameFinishedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, RenderFrameFinishedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return RenderFrameFinishedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    renderMode = RenderMode.valueOf(map.nonNullStringWithUpperCase(RENDER_MODE)),
+    needsRepaint = map.nonNullBoolean(NEEDS_REPAINT),
+    placementChanged = map.nonNullBoolean(PLACEMENT_CHANGED)
+  )
 }
 
 /**
  * Get the parsed event data for camera changed event.
  * @return a parsed CameraChangedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getCameraChangedEventData(): CameraChangedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, CameraChangedEventData::class.java)
-}
\ No newline at end of file
+  val map = data.contents as Map<String, Value>
+  return CameraChangedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END)
+  )
+}
+
+internal fun Map<String, Value>.nonNullLong(name: String): Long {
+  return this[name]!!.contents as Long
+}
+
+internal fun Map<String, Value>.nullableLong(name: String): Long? {
+  return this[name]?.contents as Long?
+}
+
+internal fun Map<String, Value>.nonNullString(name: String): String {
+  return this[name]!!.contents as String
+}
+
+internal fun Map<String, Value>.nonNullStringWithUpperCase(name: String): String {
+  return nonNullString(name).toUpperCase(Locale.US).replace("-", "_")
```

Comment:
Why is `-` replaced with `_` ? It's also hidden inside `nonNullStringWithUpperCase`, Maybe add separate extension? 
E.g. make it chained call `map.nonNullString(RENDER_MODE).upperCase().replaceDash()`

Reply:
Take `SPRITE_JSON("sprite-json")` for example, the serialize name has "-", but the enum name has "_", so we must replace it, otherwise we can't create the right enum instance.
Since the replacement only happens in this part, IMO we don't need to add another extension function for it.